kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/GodotPlugin.kt

```diff
@@ -18,24 +21,30 @@ abstract class GodotPlugin : Plugin<Project> {
     @get:Inject
     abstract val registry: ToolingModelBuilderRegistry
 
-    override fun apply(target: Project) {
-        target.checkKotlinVersionCompatibility()
+    override fun apply(target: Project) = with(target) {
+        checkKotlinVersionCompatibility()
 
-        val extension = target.extensions.create("godot", GodotExtension::class.java).also {
-            it.configureExtensionDefaults(target)
+        val extension = extensions.create("godot", GodotExtension::class.java).also {
+            it.configureExtensionDefaults(this)
         }
 
-        target.configureThirdPartyPlugins()
-        target.setupConfigurationsAndCompilations()
-        target.setupTasks()
+        configureThirdPartyPlugins()
+        setupConfigurationsAndCompilations()
+        setupTasks()
 
         // registers the tooling model builder, so it can be used by the ide plugin
-        target.afterEvaluate {
+        afterEvaluate {
+            if (godotJvmExtension.enableGodotCoroutines.get()) {
+                dependencies.add(
+                    "implementation",
+                    dependencies.create("com.utopia-rise:${godotCoroutineLibraryArtifactName}:${GodotBuildProperties.assembledGodotKotlinJvmVersion}")
+                )
+            }
             registry.register(
                 PropertiesModelBuilder(
                     isFqNameRegistrationEnabled = extension.isFqNameRegistrationEnabled.get(),
```

Comment:
Please leave the explicit target be. There were are some scoping issues before and i would not like to introduce such issues again